### DISCUSSION TITLE
nanoVNA Saver 0.5.5 keeps crashing during Resonance Analysis

### DISCUSSION THREAD

**Discussion by mitchiali** [NanoVNA-Saver/nanovna-saver#608] [General]
Connected NanoVNA-H4 with nanoVNA saver.
When i try doing a "Resonance Analysis", it crashes.

**Comment by Ho-Ro** [comment]
You're a Windows user?
Please start from a cmd window and provide the error message.

**Reply by mitchiali** [reply]
C:\Ali\nanoVNA Saver>nanovna-saver
NanoVNASaver 0.5.5

Copyright (C) 2019, 2020 Rune B. Broberg
Copyright (C) 2020ff NanoVNA-Saver Authors

This program comes with ABSOLUTELY NO WARRANTY
This program is licensed under the GNU General Public License version 3

See https://github.com/NanoVNA-Saver/nanovna-saver for further details.

Traceback (most recent call last):
  File "NanoVNASaver\Windows\AnalysisWindow.py", line 105, in runAnalysis
  File "NanoVNASaver\Analysis\ResonanceAnalysis.py", line 105, in runAnalysis
  File "NanoVNASaver\Analysis\ResonanceAnalysis.py", line 110, in do_resonance_analysis
TypeError: cannot unpack non-iterable numpy.int64 object

**Reply by zarath** [reply]
it's a bug introduced by some code cleanup... will be fixed in next release (0.6.0)
I have move the project structure to more clean src subdir layout for tox, pyscaffold etc
therefore the version jump

But I have to do some more cleanups

**Reply by zarath** [reply]
The current fix may not be as precise as the previous one regarding the interpolation of the exact frequency but also the old just delivered messured freq point and no real interpolation
(maximum error = 1 step)

**Reply by Lightwel** [reply]
Hi - same problem using the short installation routine on Manjaro :

❯ uname -r
5.15.105-1-MANJARO

❯ pip3 install https://github.com/NanoVNA-Saver/nanovna-saver/archive/refs/tags/v0.5.5.tar.gz

❯ NanoVNASaver
NanoVNASaver 0.5.5

Copyright (C) 2019, 2020 Rune B. Broberg
Copyright (C) 2020ff NanoVNA-Saver Authors

This program comes with ABSOLUTELY NO WARRANTY
This program is licensed under the GNU General Public License version 3

See https://github.com/NanoVNA-Saver/nanovna-saver for further details.

Then ran a sweep and tried Resonance Analysis.   This is the CMD line output:

Traceback (most recent call last):
  File "/home/richardhornby/.local/lib/python3.10/site-packages/NanoVNASaver/Windows/AnalysisWindow.py", line 105, in runAnalysis
    self.analysis.runAnalysis()
  File "/home/richardhornby/.local/lib/python3.10/site-packages/NanoVNASaver/Analysis/ResonanceAnalysis.py", line 105, in runAnalysis
    self.do_resonance_analysis()
  File "/home/richardhornby/.local/lib/python3.10/site-packages/NanoVNASaver/Analysis/ResonanceAnalysis.py", line 110, in do_resonance_analysis
    start, lowest, end = m
TypeError: cannot unpack non-iterable numpy.int64 object
zsh: IOT instruction (core dumped)  NanoVNASaver

Any help pls?

Tks
R

**Comment by zarath** [comment]
Shoud be posted in Issues if still existing in v0.6.8
